### PR TITLE
Add GitHub support

### PR DESCRIPTION
**Pull request by dblanchette**
Please note that `useDefaultReviewers` has no effect on GitHub. [GitHub uses the concept of Code Owners instead](https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-code-owners)

### CODE REVIEW

**Review by dreisch-coveo — APPROVED** [review]
👍

File: lib/lure/command/commandinterfaces.go [hunk 1]
```diff
@@ -20,10 +20,12 @@ type sourceControl interface {
 	Commit(string) (string, error)
 }
 
-type repository interface {
+type Repository interface {
```
Comment (AndyCoveo) [inline lib/lure/command/commandinterfaces.go:23]: Why export this type?

File: lure.go [hunk 2]
```diff
@@ -125,9 +135,23 @@ func runMain(config *project.LureConfig, auth vcs.Authentication) {
...
 func mainWithEnvironmentAuth(config *project.LureConfig) {
-	auth := vcs.UserPassAuth{
-		Username: os.Getenv("BITBUCKET_USERNAME"),
-		Password: os.Getenv("BITBUCKET_PASSWORD"),
+	var auth vcs.Authentication
+	accessToken := os.Getenv("GITHUB_ACCESS_TOKEN")
+	if accessToken != "" {
+		auth = vcs.TokenAuth{User: "x-access-token", Token: accessToken}
+	} else {
+		username := os.Getenv("GITHUB_USERNAME")
```
Comment (AndyCoveo) [inline lure.go:145]: can we move this to the provider? GetUserName, GetPassword

File: lure.go [hunk 3]
```diff
@@ -225,6 +249,12 @@ func loadConfig(filePath string) (*project.LureConfig, error) {
 	if err := json.NewDecoder(file).Decode(lureConfig); err != nil {
 		return nil, err
 	}
+	// Default value for host
+	for i, lureProject := range lureConfig.Projects {
+		if lureProject.Host == "" {
+			lureConfig.Projects[i].Host = "bitbucket"
```
Comment (AndyCoveo) [inline lure.go]: Bitbucket from the vcs constant?

File: lib/lure/repositorymanagementsystem/github.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	log.Logger.Info(fmt.Sprintf("Created Pull Request %x", *pr.Number))
+
+	return nil
+}
+
+func (gh GitHub) GetPullRequests(username string, repoSlug string, ignoreDeclinedPRs bool) ([]PullRequest, error) {
+	httpClient := gh.authentication.AuthenticateWithToken()
+	client := github.NewClient(httpClient)
+
+	options := github.PullRequestListOptions{State: "open"}
```
Comment (AndyCoveo) [inline lib/lure/repositorymanagementsystem/github.go]: if IgnoreDecliedPRs is true, we should also list declined PRs

File: lure.go [hunk 5]
```diff
@@ -84,16 +84,26 @@ func runMain(config *project.LureConfig, auth vcs.Authentication) {
...
+
+		var provider command.Repository
+		switch projectConfig.Host {
+		case vcs.GitHub:
+			provider = repository.NewGitHub(auth, projectConfig)
+		case vcs.Bitbucket:
+			provider = repository.NewBitbucket(auth, projectConfig)
+		default:
+			// host = nil
+			err = fmt.Errorf("Unknown Host '%s' - must be one of %s, %s", projectConfig.Host, vcs.GitHub, vcs.Bitbucket)
```
Comment (AndyCoveo) [inline lure.go:97]: should we leave at this point?

File: lib/lure/command/commandinterfaces.go (hunk 1 above)
Comment (dblanchette) [reply lib/lure/command/commandinterfaces.go:23]: I need it in lure.go, in this part (first line):
```var provider command.Repository
switch projectConfig.Host {
case vcs.GitHub:
	provider = repository.NewGitHub(auth, projectConfig)
case vcs.Bitbucket:
	provider = repository.NewBitbucket(auth, projectConfig)
default:
	// host = nil
	err = fmt.Errorf("Unknown Host '%s' - must be one of %s, %s", projectConfig.Host, vcs.GitHub, vcs.Bitbucket)
}

File: lure.go (hunk 5 above)
Comment (dblanchette) [reply lure.go:97]: Yes, I don't know why we don't leave when we don't know the VCS either

File: lib/lure/command/commandinterfaces.go (hunk 1 above)
Comment (AndyCoveo) [reply lib/lure/command/commandinterfaces.go:23]: hum I see, ok lets leave it like that for now.

File: lure.go (hunk 2 above)
Comment (dblanchette) [reply lure.go:145]: The problem is the auth is determined before the provider. So, it's possible, but that would necessitate a refactor.

**Pull request merged**